darts/models/regression_model.py

```diff
@@ -13,300 +13,289 @@
+        Can be used to fit any scikit-learn-like regressor class to predict the target
+        time series with lagged values.
 
-        # state
+        Parameters
+        ----------
+        lags : Union[int, list]
+            Number of lagged target values used to predict the next time step. If an integer is given
+            the last `lags` lags are used (inclusive). Otherwise a list of integers with lags is required.
+        lags_exog : Union[int, list, bool]
+            Number of lagged exogenous values used to predict the next time step. If an integer is given
+            the last `lags_exog` lags are used (inclusive). Otherwise a list of integers with lags is required.
+            If True `lags` will be used to determine lags_exog. If False, the values of all exogenous variables
+            at the current time `t`. This might lead to leakage if for predictions the values of the exogenous
+            variables at time `t` are not known.
+        model
+            A regression model that implements `fit()` and `predict()` methods.
+            Default: `sklearn.linear_model.LinearRegression(n_jobs=-1, fit_intercept=False)`
+        """
+        raise_if((lags is None) and (lags_exog is None),
+            "At least one of `lags` or `lags_exog` must be not None."
+        )
+        raise_if_not(isinstance(lags, (int, list)) or lags is None,
+            "`lags` must be of type int or list."
+        )
+        raise_if_not(isinstance(lags_exog, (int, list, bool)) or lags_exog is None,
+            "`lags_exog` must be of type int, list or bool."
+        )
+        raise_if(lags is None and lags_exog is True,
+            "`lags_exog` must not be True if `lags` is None."
+        )
+
+        if (not callable(getattr(model, "fit", None))):
+            raise_log(Exception('Provided model object must have a fit() method', logger))
+        if (not callable(getattr(model, "predict", None))):
+            raise_log(Exception('Provided model object must have a predict() method', logger))
+
+        if model is None:
+            model = LinearRegression()
+
+        self.lags = lags
+        if isinstance(self.lags, int):
+            raise_if_not(self.lags > 0, "`lags` must be strictly positive. Given: {}.".format(self.lags))
+            self.lags = list(range(1, self.lags+1))
+        elif isinstance(self.lags, list):
+            for lag in self.lags:
+                raise_if(not isinstance(lag, int) or (lag <= 0), (
+                    "Every element of `lags` must be a strictly positive integer. Given: {}.".format(self.lags))
+                )
+
+        self.lags_exog = lags_exog
+        if self.lags_exog is True:
+            self.lags_exog = self.lags[:]
+        elif self.lags_exog is False:
+            self.lags_exog = [0]
+        elif isinstance(self.lags_exog, int):
+            raise_if_not(self.lags_exog > 0, "`lags_exog` must be strictly positive. Given: {}.".format(self.lags_exog))
```

Comment:
We still need to handle the `lags_exog=False` case, so I changed it to `elif self.lags_exog is False or self.lags_exog==0`.
Maybe it would also be a good idea to **not** support `type(lags_exog)==bool`. It is somewhat confusing probably and if a user wants the same lag for exog and endog he can very easily do something like 
```python
lags = [1, 2, 3, 4]
model = RegressionModel(lags=lags, lags_exog=lags)
```

Also should we maybe rename the `lags` argument to `lags_endog` or `lags_series`? Let me know what you think.

Reply:
I agree with your point about removing the `bool` - probably clearer to just specify the lags explicitly. 
I think `lags` is already not a bad name, I would say no need to rename.